### PR TITLE
Refactor change YamlRuleSchemaMetaData same with  RuleSchemaMetaData

### PR DESCRIPTION
**Pull request by xbkaishui**
Fixes #7559.

Changes proposed in this pull request:
- change YamlRuleSchemaMetaData same with  RuleSchemaMetaData

### CODE REVIEW

File: shardingsphere-governance/shardingsphere-governance-core/shardingsphere-governance-core-config/src/test/resources/yaml/metadata.yaml [hunk 1]
```diff
@@ -30,15 +30,4 @@ configuredSchemaMetaData:
...
-          id:
-            caseSensitive: false
-            dataType: 0
-            generated: false
-            name: id
-            primaryKey: true
-        indexes:
-          primary:
-            name: PRIMARY
+    - t_user
```
Comment (tristaZero) [inline shardingsphere-governance/shardingsphere-governance-core/shardingsphere-governance-core-config/src/test/resources/yaml/metadata.yaml:33]: Please add a blank line under this line.

File: shardingsphere-governance/shardingsphere-governance-core/shardingsphere-governance-core-common/src/main/java/org/apache/shardingsphere/governance/core/yaml/config/metadata/YamlRuleSchemaMetaData.java [hunk 2]
```diff
@@ -32,5 +33,5 @@
 
     private YamlSchemaMetaData configuredSchemaMetaData;
 
-    private Map<String, YamlSchemaMetaData> unconfiguredSchemaMetaDataMap;
+    private Map<String, Collection<String>> unconfiguredSchemaMetaDataMap;
```
Comment (tristaZero) [inline shardingsphere-governance/shardingsphere-governance-core/shardingsphere-governance-core-common/src/main/java/org/apache/shardingsphere/governance/core/yaml/config/metadata/YamlRuleSchemaMetaData.java:36]: This PR looks great. However, I found another issue relevant to what you modified. 

```java
// CreateTableStatementMetaDataRefreshStrategy

private void refreshUnconfiguredMetaData(final ShardingSphereMetaData metaData, 
                                             final DatabaseType databaseType, final Map<String, DataSource> dataSourceMap, final String tableName) throws SQLException {
        for (Entry<String, DataSource> entry : dataSourceMap.entrySet()) {
            Optional<TableMetaData> tableMetaData = TableMetaDataLoader.loadWithoutColumnMetaData(entry.getValue(), tableName, databaseType.getName());
            if (tableMetaData.isPresent()) {
                refreshUnconfiguredMetaData(metaData, tableName, entry.getKey());
                return;
            }
        }
    }

```
`loadWithoutColumnMetaData` is supposed to be simplified, IMO. What do you think?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-governance/shardingsphere-governance-core/shardingsphere-governance-core-config/src/test/resources/yaml/metadata.yaml (hunk 1 above)
Comment (xbkaishui) [reply shardingsphere-governance/shardingsphere-governance-core/shardingsphere-governance-core-config/src/test/resources/yaml/metadata.yaml]: done

File: shardingsphere-governance/shardingsphere-governance-core/shardingsphere-governance-core-common/src/main/java/org/apache/shardingsphere/governance/core/yaml/config/metadata/YamlRuleSchemaMetaData.java (hunk 2 above)
Comment (xbkaishui) [reply shardingsphere-governance/shardingsphere-governance-core/shardingsphere-governance-core-common/src/main/java/org/apache/shardingsphere/governance/core/yaml/config/metadata/YamlRuleSchemaMetaData.java:36]: done

**Review by tristaZero — APPROVED** [review]
Looks rather concise!

**Pull request merged**